spring-boot-admin-client/src/main/java/de/codecentric/boot/admin/client/registration/DefaultApplicationFactory.java

```diff
@@ -84,7 +84,22 @@ protected String getServiceUrl() {
             return instance.getServiceUrl();
         }
 
-        return UriComponentsBuilder.fromUriString(getServiceBaseUrl()).path("/").toUriString();
+        return UriComponentsBuilder.fromUriString(getServiceBaseUrl())
+                                   .path("/")
+                                   .path(getServicePath())
+                                   .toUriString();
+    }
+
+    protected String getServicePath() {
+        if (instance.getServicePath() != null) {
+            return instance.getServicePath();
+        }
+
+        return getServerContextPath();
+    }
+
+    protected String getServerContextPath() {
+        return server.getServlet().getContextPath();
```

Comment:
Imho `server.getServlet().getContextPath();` should be moved to the `ServletApplicationFactory`and replaced with `/`.